### PR TITLE
feat: add C implementation for `math/base/special/cospi`

### PR DESCRIPTION
**Pull request by gunjjoshi**
## Description

> What is the purpose of this pull request?

This pull request:

-   adds C implementation for [`math/base/special/cospi`](https://github.com/stdlib-js/stdlib/tree/32558fd132a6a004069de40b6e25bdb2e5521a07/lib/node_modules/%40stdlib/math/base/special/cospi).

```c
double stdlib_base_cospi( const double x );
```

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves a part of #649.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/cospi/src/main.c [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+*/
+
+/*
+* Notes:
+*	=> cos(-x) = cos(x)
+*	=> sin(-x) = -sin(x)
+*	=> cos(π/2) = 0
+*	=> cos(0) = 1
+*	=> cos(π) = -1
+*/
```
Comment (gunjjoshi) [inline lib/node_modules/@stdlib/math/base/special/cospi/src/main.c]: I'll change this based on our conclusion at https://github.com/stdlib-js/stdlib/pull/2610#discussion_r1679467938

**Review by Planeshifter — APPROVED** [review]
Looks good; will merge once we have resolved open discussion and potentially made changes to the comment.

File: lib/node_modules/@stdlib/math/base/special/cospi/src/main.c [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+/*
+* Notes:
+*	=> cos(-x) = cos(x)
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/cospi/src/main.c]: This note also contains TABs.

Comment (gunjjoshi) [reply lib/node_modules/@stdlib/math/base/special/cospi/src/main.c]: I've changed it now.

Comment (Planeshifter) [reply lib/node_modules/@stdlib/math/base/special/cospi/src/main.c]: Thanks, @gunjjoshi!

**Pull request merged**